osu.Game/Database/StandardisedScoreMigrationTools.cs

```diff
@@ -185,6 +187,36 @@ public static long GetOldStandardised(ScoreInfo score)
             return (long)Math.Round((1000000 * (accuracyPortion * accuracyScore + (1 - accuracyPortion) * comboScore) + bonusScore) * modMultiplier);
         }
 
+        public static long ChangeComboRatio(ScoreInfo score, double oldComboPortion, double newComboPortion)
+        {
+            // Assume the incoming score already has a total score in the newest standardised calculation method.
+            double totalScore = score.TotalScore;
+
+            // Undo mod multiplier to leave us with combo and accuracy portions.
+            double scoreMultiplier = 1;
+            foreach (var m in score.Mods)
+                scoreMultiplier *= m.ScoreMultiplier;
+
+            totalScore /= scoreMultiplier;
+
+            var processor = runBestCaseScoreProcessor(score);
+
+            // Remove the accuracy and bonus portions (which we can recalculate)
+            totalScore -= processor.AccuracyPortion * Math.Pow(processor.Accuracy.Value, 10) * (1000000 * (1 - oldComboPortion));
+            totalScore -= processor.BonusPortion;
```

Comment:
This is also slightly inaccurate as it assumes the player got all possible points from the bonus portion which may not be the case. It'll have the end effect of slightly over- or underestimating the score due to too much score being attributed to the combo portion. I guess it is not really possible to do better than this from the data that we have, though.